.github/workflows/release-plugins.yaml

```diff
@@ -0,0 +1,114 @@
+name: Backstage knative-event-mesh-backend
+
+on:
+  pull_request:
+    branches:
+      - 'release-*'
+    paths:
+      - 'backstage/plugins/knative-event-mesh-backend/**'
+  push:
+    branches:
+      - 'release-*'
+    paths:
+      - 'backstage/plugins/knative-event-mesh-backend/**'
+
+jobs:
+  release:
+    name: Release
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Setup release name for release branch push
+        if: github.event_name == 'push'
+        run: |
+          BRANCH=$(git rev-parse --abbrev-ref HEAD) #e.g. release-v1.33
+          echo "Release branch: ${BRANCH}"
+          GO111MODULE=off go get -u github.com/openshift-knative/hack/cmd/sobranch
+          SO_RELEASE_NAME=$( $(go env GOPATH)/bin/sobranch --upstream-version "${BRANCH}") # input format: "release-v1.11" or "release-1.11" or "v1.11" or "1.11"
+          # SO_RELEASE_NAME will be something like "release-1.33"
+          echo "SO_RELEASE_NAME: ${SO_RELEASE_NAME}"
+          # split the release name to get the version number
+          ## e.g. release-1.33 -> v1.33
+          BASE_RELEASE_VERSION=${SO_RELEASE_NAME#*-}
+          echo "BASE_RELEASE_VERSION: ${BASE_RELEASE_VERSION}"
+          DATE_TIME=$(date +"%Y%m%d%H%M%S")
+          RELEASE_NAME="${BASE_RELEASE_VERSION}.0-${DATE_TIME}"
```

Comment:
can we append the short commit sha instead of the date?

Reply:
I think commit hash alone is not good, but I can add it to the end like `release-1.2.3-202401010101-deadbeef`

Reply:
why is not good?

Reply:
Also the `-` is an indicator of pre-releases, I think what we want is the build identifier with a `+`

Reply:
I'm also wondering if we should increment the patch version versus pre-releases or build numbers

Reply:
Since I'm not sure how that works when using something like `~1.33.0`

Reply:
What I understand is that when you use `+`:
- `1.0.0+20240101-deadbeef` will be resolved to be same version as `1.0.0+2099....-...`, which we don't want here.

When you use `-`, `1.0.0+20240101-deadbeef` < `1.0.0+2099....-...` will be true, as 2099 is alphabetically greater.

Reply:
Well following might work:
- In release branches, store the actual version (like 1.12.0), not 0.0.0 in `package.json`
- After a commit, increase the version in package.json, then do the release, then commit the version to the branch

Reply:
It's possible, I'm just not sure how you would do this

> After a commit, increase the version in package.json, then do the release

How would you distinguish which commit to release vs not ?

Reply:
I think we can also list versions and figure out what's the "next patch for a major.minor"

Reply:
Yeah, all commits in a release branch would cause a new package being published.

Reply:
```
BASE_RELEASE_NAME="0.1"; npm view @knative-extensions/plugin-knative-event-mesh-backend versions --json | grep ${BASE_RELEASE_NAME} | tail -1 | tr -d '"' | awk -F \. '{print $3+1}'
```

This will give us the next patch version for `@knative-extensions/plugin-knative-event-mesh-backend` package given the `BASE_RELEASE_NAME` being `0.1`

Reply:
Ok, what can go wrong, right? :D Let me incorporate this

Reply:
A lot can go wrong :) but I think it's a little better

for sure we need to add the concurrency https://docs.github.com/en/actions/using-jobs/using-concurrency as a follow up